modules/pricer/src/main/java/com/opengamma/strata/pricer/sensitivity/CurveSensitivityUtils.java

```diff
@@ -0,0 +1,160 @@
+      List<CurveParameterMetadata> lm = m.getParameterMetadata().get();
+      for (int loopnode = 0; loopnode < sa.size(); loopnode++) {
+        ArgChecker.isTrue(lm.get(loopnode) instanceof DatedCurveParameterMetadata,
+            "re-bucketing require sensitivity date");
+        DatedCurveParameterMetadata dpm = (DatedCurveParameterMetadata) lm.get(loopnode);
+        LocalDate nodeDate = dpm.getDate();
+        rebucketingArray(targetDates, nbBuckets, rebucketedSensitivityAmounts, sa.get(loopnode), nodeDate);
+      }
+      CurveCurrencyParameterSensitivity rebucketedSensitivity =
+          CurveCurrencyParameterSensitivity.of(
+              DefaultCurveMetadata.builder().curveName(s.getCurveName()).parameterMetadata(pmdTarget).build(),
+              s.getCurrency(), DoubleArray.ofUnsafe(rebucketedSensitivityAmounts));
+      sensitivityTarget.add(rebucketedSensitivity);
+    }
+    return CurveCurrencyParameterSensitivities.of(sensitivityTarget);
+  }
+
+
+  /**
+   * Re-buckets a {@link CurveCurrencyParameterSensitivities} to a given set of dates. 
+   * <p>
+   * The list of dates must be sorted in chronological order. All sensitivities are re-bucketed to the same date list.
+   * The re-bucketing is done by linear weighting on the number of days, i.e. the sensitivities for dates outside the 
+   * extremes are fully bucketed to the extremes and for date between two re-bucketing dates, the weight on the start 
+   * date is the number days between end date and the date re-bucketed divided by the number of days between the 
+   * start and the end.
+   * The input sensitivity should have a {@link DatedCurveParameterMetadata} for each sensitivity. 
+   * 
+   * @param sensitivities  the input sensitivities
+   * @param targetDates  the list of dates for the re-bucketing
+   * @param sensitivityDate  the date for which the sensitivities are valid
+   * @return the sensitivity after the re-bucketing
+   */
+  public static CurveCurrencyParameterSensitivities linearRebucketing(
+      CurveCurrencyParameterSensitivities sensitivities,
+      List<LocalDate> targetDates,
+      LocalDate sensitivityDate) {
+    int nbBuckets = targetDates.size();
+    List<CurveParameterMetadata> pmdTarget = new ArrayList<>();
+    for (int i = 0; i < nbBuckets; i++) {
+      pmdTarget.add(SimpleCurveNodeMetadata.of(targetDates.get(i), targetDates.get(i).toString()));
+    }
+    ImmutableList<CurveCurrencyParameterSensitivity> listSensi = sensitivities.getSensitivities();
+    List<CurveCurrencyParameterSensitivity> sensitivityTarget = new ArrayList<>();
+    for (CurveCurrencyParameterSensitivity s : listSensi) {
+      double[] rebucketedSensitivityAmounts = new double[nbBuckets];
+      CurveMetadata m = s.getMetadata();
+      DoubleArray sa = s.getSensitivity();
+      ArgChecker.isTrue(m.getParameterMetadata().isPresent(), "parameter metadata must be present");
```

Comment:
This is more idiomatic:

          List<CurveParameterMetadata> lm = m.getParameterMetadata().orElseThrow(() -> new IllegalArgumentException(...));